process/factory/interceptorscontainer/baseInterceptorsContainerFactory.go

```diff
@@ -639,107 +680,162 @@ func (bicf *baseInterceptorsContainerFactory) generatePeerAuthenticationIntercep
 			AntifloodHandler:     bicf.antifloodHandler,
 			WhiteListRequest:     bicf.whiteListHandler,
 			PreferredPeersHolder: bicf.preferredPeersHolder,
-			CurrentPeerId:        bicf.messenger.ID(),
+			CurrentPeerId:        bicf.mainMessenger.ID(),
 		},
 	)
 	if err != nil {
 		return err
 	}
 
-	interceptor, err := bicf.createTopicAndAssignHandler(identifierPeerAuthentication, mdInterceptor, true)
+	err = createTopicAndAssignHandlerOnMessenger(identifierPeerAuthentication, mdInterceptor, true, bicf.mainMessenger)
 	if err != nil {
 		return err
 	}
 
-	return bicf.container.Add(identifierPeerAuthentication, interceptor)
+	return bicf.container.Add(identifierPeerAuthentication, mdInterceptor)
 }
 
-//------- Heartbeat interceptor
+//------- Heartbeat interceptors
 
-func (bicf *baseInterceptorsContainerFactory) generateHeartbeatInterceptor() error {
-	// TODO[Sorin]: we'll need separate interceptors for full archive heartbeats and peer authentications
+func (bicf *baseInterceptorsContainerFactory) generateMainHeartbeatInterceptor() error {
 	shardC := bicf.shardCoordinator
 	identifierHeartbeat := common.HeartbeatV2Topic + shardC.CommunicationIdentifier(shardC.SelfId())
 
+	interceptor, err := bicf.createOneHeartbeatV2Interceptor(identifierHeartbeat, bicf.dataPool.Heartbeats(), bicf.mainPeerShardMapper)
+	if err != nil {
+		return err
+	}
+
+	err = createTopicAndAssignHandlerOnMessenger(identifierHeartbeat, interceptor, true, bicf.mainMessenger)
+	if err != nil {
+		return err
+	}
+
+	return bicf.container.Add(identifierHeartbeat, interceptor)
+}
+
+func (bicf *baseInterceptorsContainerFactory) generateFullArchiveHeartbeatInterceptor() error {
+	if bicf.nodeOperationMode != p2p.FullArchiveMode {
+		return nil
+	}
+
+	shardC := bicf.shardCoordinator
+	identifierHeartbeat := common.FullArchiveTopicPrefix + common.HeartbeatV2Topic + shardC.CommunicationIdentifier(shardC.SelfId())
```

Comment:
if we get rid of this const (FullArchiveTopicPrefix) we can merge `generateMainHeartbeatInterceptor` with `generateFullArchiveHeartbeatInterceptor`

Reply:
updated